System/Random.hs

```diff
@@ -68,57 +66,34 @@
 -- >>> fst $ stepGen $ snd $ stepGen (PCGen 17 29)
 -- 3288430965
 --
+-- Once implemented an instance of `RandomGen` can be created:
+--
 -- >>> :{
 -- instance RandomGen PCGen where
---   next g = (fromIntegral y, h)
---     where
---       (y, h) = stepGen g
+--   genWord32 = stepGen
 --   split _ = error "This PRNG is not splittable"
 -- :}
 --
--- Importantly, this implementation will not be as efficient as it
--- could be because the random values are converted to 'Integer' and
--- then to desired type.
---
--- Instead we should define (where e.g. @unBuildWord32 :: Word32 ->
--- (Word16, Word16)@ is a function to pull apart a 'Word32' into a
--- pair of 'Word16'):
---
--- >>> newtype PCGen' = PCGen' { unPCGen :: PCGen }
+-- Note, that depending on thow many bits of randomness your RNG produces in one iteration
+-- you might need to implement a different function in the `RandomGen` class.
 --
--- >>> let stepGen' = second PCGen' . stepGen . unPCGen
---
--- >>> :{
--- instance RandomGen PCGen' where
---   genWord8 = first fromIntegral . stepGen'
---   genWord16 = first fromIntegral . stepGen'
---   genWord32 = stepGen'
---   genWord64 g = (buildWord64 x y, g'')
---       where
---       (x, g') = stepGen' g
---       (y, g'') = stepGen' g'
---       buildWord64 w0 w1 = ((fromIntegral w1) `shiftL` 32) .|. (fromIntegral w0)
--- :}
+-- Once implemented, every pure generator can be used with `MonadRandom` by the means of
+-- `PureGen` and state transformers. See an example below of such use case.
 --
 -- [/Example for RNG Users:/]
 --
--- Suppose you want to simulate rolls from a dice (yes I know it's a
+-- Suppose you want to simulate a number of rolls from a dice (yes I know it's a
 -- plural form but it's now common to use it as a singular form):
 --
 -- >>> :{
--- let randomListM :: (MonadRandom g m, Num a, Uniform a) => g -> Int -> m [a]
---     randomListM gen n = replicateM n (uniform gen)
--- :}
---
--- >>> :{
--- let rolls :: [Word32]
---     rolls = runGenState_
+-- let rolls :: Int -> [Word]
+--     rolls n = runGenState_
 --               (PCGen 17 29)
---               (\g -> randomListM g 10 >>= \xs -> return $ map ((+1) . (`mod` 6)) xs)
```

Comment:
I added `uniformListM` as an exported function, it seems like it could be useful

Reply:
I'm a fan of keeping the API surface small, particularly given that this is a core package (for some definition of "core"). "seems like it could be useful" is not strong enough IMO. I would prefer it if we didn't export `uniformListM`.

Reply:
There is `randoms`, so I though it only made sense to add a monadic version of it. Also I think generating a list of values is gonna be an extremely common operation. Hell it even creeped up in documentation ;)

Reply:
It seems to be have been removed from the example - fine with me

Reply:
I removed it because using `uniformR` is a correct way of generating random die throws, rather than `mod 6`, cause it is biased that way.